libbeat/processors/add_kubernetes_metadata/kubernetes.go

```diff
@@ -51,6 +55,16 @@ func init() {
 	Indexing.AddMatcher(FieldFormatMatcherName, NewFieldFormatMatcher)
 }
 
+func isKubernetesAvailable(client k8s.Interface) bool {
+	info, err := client.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		logp.Debug("kubernetes", "%v: could not detect kubernetes env", "add_kubernetes_metadata")
```

Comment:
I haven't followed this path, but I'm not sure if the user gets a clear message when kubernetes is not available.
If there is a message that states it clearly dismiss my comment.

Reply:
Since we want to have Kubernetes processor running silently (without enriching events) even if Kubernetes is not available, we just log in debug mode that Kubernetes was not detected (or that the client could not be created, if there is no actual kubeconfig in default path `~/.kube/config`). So now that we have the processor to be enabled by default we have two possible cases:
1. Kubernetes environment could not be verified, either because it does not exist or because it is not properly configured. In this situation we initialize a "silent" processor and just state that in the logs.
2. Kubernetes env was verified in the system and we have a complete processor enriching the events.